Exercise_2/Adapters/BillAdapter.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+
+using Android.App;
+using Android.Content;
+using Android.OS;
+using Android.Runtime;
+using Android.Support.V7.Widget;
+using Android.Views;
+using Android.Widget;
+using Exercise_2.Models;
+
+namespace Exercise_2.Adapters
+{
+    internal class BillAdapter : RecyclerView.Adapter
+    {
+        public List<Order> Orders { get; set; }
+
+        public BillAdapter(List<Order> orders)
+        {
+            Orders = orders;
+        }
+        public override void OnBindViewHolder(RecyclerView.ViewHolder holder, int position)
+        {
+
+            if (!(holder is BillItemViewHolder viewHolder)) return;
+            viewHolder.Name.Text = Orders[position].Product.Name;
```

Comment:
You should move the logic code to the class `BillItemViewHolder` so here you can do like this
```C#

public override void OnBindViewHolder(BillItemViewHolder holder, int position)
{
      holder.BillItem = Orders[position];
}

```